### PR TITLE
arch: cleanup EXCEPTION_STACK_TRACE dependencies

### PR DESCRIPTION
**Pull request by ycsin**
> [!NOTE]
> Please review #73288 instead, thanks.

Move X86's dependencies from `EXCEPTION_STACK_TRACE` to `X86_EXCEPTION_STACK_TRACE`

### CODE REVIEW

File: arch/x86/core/Kconfig.ia32 [hunk 1]
```diff
@@ -167,6 +167,7 @@ config X86_EXCEPTION_STACK_TRACE
 	bool
 	default y
 	depends on EXCEPTION_STACK_TRACE
+	depends on DEBUG_INFO
```
Comment (dcpleung) [inline arch/x86/core/Kconfig.ia32:170]: I think a `select DEBUG_INFO` would be better here.

Comment (ycsin) [reply arch/x86/core/Kconfig.ia32]: Fixed

File: arch/x86/core/Kconfig.intel64 [hunk 2]
```diff
@@ -32,8 +32,10 @@ config X86_EXCEPTION_STACK_SIZE
 config X86_EXCEPTION_STACK_TRACE
 	bool
 	default y
+	select DEBUG_INFO
+	select NO_OPTIMIZATIONS
```
Comment (ycsin) [inline arch/x86/core/Kconfig.intel64:36]: @dcpleung I also added a commit to select `NO_OPTIMIZATIONS` here instead of depending on it

Comment (ycsin) [reply arch/x86/core/Kconfig.intel64]: That was a bad idea, removed

**Pull request closed without merging**